modules/compute-vm/resource-policies.tf

```diff
@@ -142,7 +142,7 @@ resource "google_compute_disk_resource_policy_attachment" "attached" {
   )
   disk = (
     each.value.source_type == "attach"
-    ? each.value.source
+    ? element(split("/", each.value.source), length(split("/", each.value.source)) - 1)
```

Comment:
maybe `reverse(split("/", each.value.source))[0]`?

Reply:
Yeah, you've right, it's cleaner. Thanks :)

Reply:
I want to double check on that, in README.md we have:
> - `source_type = "attach"` can be used for both instances and templates to attach an existing disk, set source to the name (for zonal disks) or self link (for regional disks) of the existing disk to attach; no disk will be created

I think this needs fixing README.md too (and removing regional examples)

The other approach (preferred) is to actually implement attachments for regional disks using `google_compute_region_disk_resource_policy_attachment` and keeping the current semantics of the interface.

@ludoo WDYT?

Reply:
attach is used in the module to flag disk creation and management, let's have a discussion because I don't think this PR is doing what you think :)